### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Josh-Kaiser-OSU**
To get user state:
``` JavaScript
import { useAuth } from ''../context/AuthContext";
let userObject = useAuth();
```

To set user state:
``` JavaScript
import { useAuthUpdate } from ''../context/AuthContext";
let updateAuth = useAuthUpdate();
updateAuth(new_auth_object);
```

Todo:
- Connect auth to user collection in db.
- Create a new user when a new sign in occurs.
- Fetch user info from db on sign in.
- Make the useAuth object's shape consistent with user schema in db.
- Allow user to sign in with email and password and GitHub account.
- Allow the page to be refreshed without signing out the user.

### CODE REVIEW

File: portfolio-project-matching-app/Firebase/auth.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { getAuth, signInWithPopup, GoogleAuthProvider, onAuthStateChanged, signOut, auth } from "firebase/auth";
+import { useAuthUpdate } from '../context/AuthContext';
+import { firebaseApp } from "./clientApp.ts";
```
Comment (kiserd) [inline portfolio-project-matching-app/Firebase/auth.js:3]: Nice work here. I was dumb enough to plug all the Firebase Firestore Database stuff directly into `clientApp.ts`. Either too dumb, or just lacked the perspective when I was getting things setup. I should probably move the DB stuff into a dedicated file like you did.

File: portfolio-project-matching-app/components/NewLogin.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return (
+        <div>
+            { uid.user ?
+                    <SignOutButton />
+                :
+                    <SignInButton />
+            }
+        </div>
+    )
+}
```
Comment (kiserd) [inline portfolio-project-matching-app/components/NewLogin.js:42]: Man. This sure as hell simplifies the login/signup process. I look forward to watching your video.

**Review by kiserd — APPROVED** [review]
Looks awesome. Of course, I will need to review your video before any real sense of intuition begins to set.

**Review by ylijokic — APPROVED** [review]
Awesome! Thanks for putting this together. The videos were very helpful!

**Pull request merged**